packages/flutter/lib/src/material/animated_raster.dart

```diff
@@ -0,0 +1,202 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// TODO(jonahwilliams): this is temporarily a private library until we determine
+// a reasonable public API.
+part of page_transitions_theme;
+
+// Foundational widgets for the rasterized zoom page transition.
+
+/// A delegate that controls what is painted during an animation in the [_AnimatedRaster]
+/// widget.
+@immutable
+abstract class _AnimatedRasterDelegate {
+  /// const constructor so that subclasses can be const.
+  const _AnimatedRasterDelegate();
+
+  /// Whether [paint] will paint anything on the current frame.
+  ///
+  /// This is used to optimize painting and delay rasterization if a child
+  /// would not be painted. One example would be an animated opacity, if the
+  /// first frame is fully transparent then rasterization can be delayed until
+  /// the second frame.
+  @protected
+  bool useRaster(Animation<double> animation) => true;
+
+  /// Paint the children of this render object.
+  ///
+  /// The provided [image] contains a GPU resident texture of the children of
+  /// this render object.
+  ///
+  /// The [pixelRatio] is the ratio of pixels in [image] to logical pixels.
+  @protected
+  void paintRaster(PaintingContext context, ui.Image image, double pixelRatio, Animation<double> animation);
+
+  /// Paint the children of this render object.
+  @protected
+  void paint(PaintingContext context, Animation<double> animation, Rect area, PaintingContextCallback callback);
+}
+
+/// A widget that replaces [child] with a rasterized version while an animation is active.
```

Comment:
> [...] while an animation is active.

Technically, it's up to the delegate to decide when to require rasterization, no? And it could be the other way around: only when the animation is not active, if one wanted...